File: CCVTAC/CCVTAC.Console/ExternalUtilities/ToolSettings.cs
namespace CCVTAC.Console.ExternalUtilties;

/// <summary>
/// Settings to govern the behavior of an external program.
/// </summary>
/// <param name="Description">A brief summary of what the program will do, phrased as a noun.</param>
/// <param name="ProgramName">The name of the utility to be run.</param>
/// <param name="Args">All arguments to be passed to the external utility.</param>
/// <param name="WorkingDirectory"></param>
public sealed record UtilitySettings(
    string Description,
    string ProgramName,
    string Args,
    string WorkingDirectory
);

File: src/CCVTAC.Main/ExternalTools/ToolSettings.fs
namespace CCVTAC.Main.ExternalTools

/// Settings to govern the behavior of an external program.
type ToolSettings = private {
    CommandWithArgs: string
    WorkingDirectory: string
}

module ToolSettings =
    let create commandWithArgs workingDirectory =
        { CommandWithArgs = commandWithArgs
          WorkingDirectory = workingDirectory }

